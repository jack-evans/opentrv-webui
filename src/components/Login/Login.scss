
.Login {
  height: 100vh;

  .Login__content {
    align-items: center;
    display: inline-block;
    justify-content: center;
    width: 500px;

    .Login__title {
      padding:40px 0 30px 0;
    }

    .Login__input {

      .Login__email-input {
        margin-right: unset;
        padding-bottom: 20px;
      }

      .Login__password-input {
        padding-top: 10px;
      }
    }

    .Login__buttons {
      padding-top: 30px;
    }
  }
}