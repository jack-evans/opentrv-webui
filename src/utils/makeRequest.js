
export default function makeRequest (apiPath, options) {
  options.mode = 'cors'
  if (options.headers) {
    options.headers['Content-Type'] = 'application/json'
  } else {
    options.headers = {
      'Content-Type': 'application/json'
    }
  }

  return global.fetch(`http://localhost:3001${apiPath}`, options)
    .then(internal.checkStatus)
    .then(internal.parseJSON)
}

function checkStatus (response) {
  if (response.status >= 200 && response.status < 300) {
    return response
  } else {
    let error = new Error(response.statusText)
    error.response = response
    throw error
  }
}

function parseJSON (response) {
  return response.json()
}

export const internal = {
  checkStatus: checkStatus,
  parseJSON: parseJSON
}
