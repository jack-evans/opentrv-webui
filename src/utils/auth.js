import makeRequest from './makeRequest'
import decode from 'jwt-decode'

export function isAuthenticated () {
  const token = getToken()
  if (!token) {
    return false
  } else {
    const decodedToken = decode(token)
    const date = new Date(0)
    date.setUTCSeconds(decodedToken.exp)
    if (date < new Date()) {
      return false
    } else {
      return true
    }
  }
}

export function loginUser (user) {
  let url = '/api/v1/user/login'
  let options = {
    method: 'POST',
    json: true,
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(user)
  }

  return makeRequest(url, options)
    .then(res => {
      setToken(res.token)
      return Promise.resolve()
    })
}

export function setToken (token) {
  window.localStorage.setItem('token', token)
}

export function getToken () {
  return window.localStorage.getItem('token')
}

export function logout () {
  window.localStorage.removeItem('token')
}
