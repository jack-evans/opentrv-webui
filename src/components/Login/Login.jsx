import React, { Component } from 'react'
import { Button, Modal, TextInput } from 'carbon-components-react'
import { isAuthenticated, loginUser } from '../../utils/auth'

class Login extends Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      invalid: {},
      failedLogin: false
    }
    this.errorContent = (<div />)

    this.handleLoginOnClick = this.handleLoginOnClick.bind(this)
    this.handleEmailOnChange = this.handleEmailOnChange.bind(this)
    this.handlePasswordOnChange = this.handlePasswordOnChange.bind(this)
    this.handleModalOnClick = this.handleModalOnClick.bind(this)
    this.handleKeyPress = this.handleKeyPress.bind(this)
  }

  handleLoginOnClick () {
    let user = {
      email: this.state.email,
      password: this.state.password
    }

    return loginUser(user)
      .then(() => {
        window.location.assign('/overview')
      })
      .catch(() => {
        this.setState({failedLogin: true})
      })
  }

  handleKeyPress (event) {
    if (event.key === 'Enter') {
      this.handleLoginOnClick()
    }
  }

  handleEmailOnChange (event) {
    this.setState({email: event.target.value})
  }

  handlePasswordOnChange (event) {
    this.setState({password: event.target.value})
  }

  handleModalOnClick () {
    this.setState({failedLogin: false})
  }

  render () {
    if (isAuthenticated()) {
      window.location.assign('/overview')
      return null
    } else {
      return (
        <div className='Login'>
          <div className='Login__content'>
            <div className='Login__title'>
              <h1>Login to OpenTRV</h1>
            </div>
            <div className='Login__input'>
              <div className='Login__email-input'>
                <TextInput
                  id='login-email'
                  labelText='Email'
                  placeholder='Enter email address'
                  type='email'
                  required
                  onChange={this.handleEmailOnChange}
                />
              </div>
              <div className='Login__password-input'>
                <TextInput
                  id='login-password'
                  labelText='Password'
                  placeholder='Enter password'
                  type='password'
                  required
                  onChange={this.handlePasswordOnChange}
                  onKeyPress={this.handleKeyPress}
                />
              </div>
            </div>
            <div className='Login__buttons'>
              <Button
                style={{width: '120px'}}
                kind='secondary'
                className='Login__back-button'
                href='/'
              >
                Back
              </Button>
              <Button
                style={{width: '120px'}}
                className='Login__login-button'
                onClick={this.handleLoginOnClick}
              >
                Login
              </Button>
            </div>
            <Modal
              className='Login__failed-modal'
              open={this.state.failedLogin}
              passiveModal
              modalHeading='Login Failed'
              onRequestClose={this.handleModalOnClick}
            >
              <p>Failed to Login to OpenTRV.</p>
              <p>Please check your email and password was entered correctly.</p>
              <p>If problem persists please contact support</p>
            </Modal>
          </div>
        </div>
      )
    }
  }
}

export default Login
