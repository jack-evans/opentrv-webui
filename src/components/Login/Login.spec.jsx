import React from 'react'
import Login from './Login'
import { setAuthenticated } from '../../utils/auth'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Login.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<Login />)).toMatchSnapshot()
  })

  describe('when the user is already authenticated', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(true)
      window.location.assign = jest.fn()
      wrapper = shallow(<Login />)
    })

    afterEach(() => {
      jest.resetAllMocks()
      wrapper.unmount()
    })

    it('calls window.location.assign', () => {
      expect(window.location.assign).toHaveBeenCalledTimes(1)
    })

    it('calls window.location.assign with "/overview"', () => {
      expect(window.location.assign).toHaveBeenCalledWith('/overview')
    })
  })

  describe('when the user enters their email into the email input', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(false)
      wrapper = mount(<Login />)
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the state to equal what the user has entered', () => {
      wrapper.find('#login-email').at(1).simulate('change', { target: { value: 'john.doe@example.com' } })
      expect(wrapper.state().email).toEqual('john.doe@example.com')
    })
  })

  describe('when the user enters their password into the password input', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(false)
      wrapper = mount(<Login />)
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the state to equal what the user has entered', () => {
      wrapper.find('#login-password').at(1).simulate('change', { target: { value: 'password' } })
      expect(wrapper.state().password).toEqual('password')
    })
  })

  describe('when a valid user attempts to login', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(false)
      window.location.assign = jest.fn()
      wrapper = mount(<Login />)
      wrapper.find('#login-email').at(1).simulate('change', { target: { value: 'john.doe@example.com' } })
      wrapper.find('#login-password').at(1).simulate('change', { target: { value: 'password' } })
      wrapper.find('.Login__login-button').at(0).simulate('click')
    })

    afterEach(() => {
      jest.resetAllMocks()
      wrapper.unmount()
    })

    it('calls window.location.assign', () => {
      expect(window.location.assign).toHaveBeenCalledTimes(1)
    })

    it('calls window.location.assign with "/overview"', () => {
      expect(window.location.assign).toHaveBeenCalledWith('/overview')
    })
  })

  describe('when a invalid user attempts to login', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(false)
      window.location.assign = jest.fn()
      wrapper = mount(<Login />)
      wrapper.find('#login-email').at(1).simulate('change', { target: { value: 'bad.person@example.com' } })
      wrapper.find('#login-password').at(1).simulate('change', { target: { value: 'password' } })
      wrapper.find('.Login__login-button').at(0).simulate('click')
    })

    afterEach(() => {
      jest.resetAllMocks()
      wrapper.unmount()
    })

    it('does not call window.location.assign', () => {
      expect(window.location.assign).toHaveBeenCalledTimes(0)
    })

    it('sets the failedLogin state to true', () => {
      expect(wrapper.state().failedLogin).toEqual(true)
    })

    it('displays the modal', () => {
      let modal = wrapper.find('.Login__failed-modal').at(0).instance()
      expect(modal.props.open).toEqual(true)
    })

    describe('when the modal is displayed', () => {
      it('closes when the cross is clicked', () => {
        wrapper.find('.bx--modal-close').at(0).simulate('click')
        expect(wrapper.state().failedLogin).toEqual(false)
      })
    })
  })
})
