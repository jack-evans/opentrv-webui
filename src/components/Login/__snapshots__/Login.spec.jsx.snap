// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Login__content"
      >
        <div
          className="Login__title"
        >
          <h1>
            Login to OpenTRV
          </h1>
        </div>
        <div
          className="Login__input"
        >
          <div
            className="Login__email-input"
          >
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="login-email"
              invalid={false}
              invalidText=""
              labelText="Email"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Enter email address"
              required={true}
              type="email"
            />
          </div>
          <div
            className="Login__password-input"
          >
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="login-password"
              invalid={false}
              invalidText=""
              labelText="Password"
              onChange={[Function]}
              onClick={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter password"
              required={true}
              type="password"
            />
          </div>
        </div>
        <div
          className="Login__buttons"
        >
          <Button
            className="Login__back-button"
            disabled={false}
            href="/"
            iconDescription="Provide icon description if icon is used"
            kind="secondary"
            small={false}
            style={
              Object {
                "width": "120px",
              }
            }
            tabIndex={0}
            type="button"
          >
            Back
          </Button>
          <Button
            className="Login__login-button"
            disabled={false}
            iconDescription="Provide icon description if icon is used"
            kind="primary"
            onClick={[Function]}
            small={false}
            style={
              Object {
                "width": "120px",
              }
            }
            tabIndex={0}
            type="button"
          >
            Login
          </Button>
        </div>
        <Modal
          className="Login__failed-modal"
          iconDescription="close the modal"
          modalHeading="Login Failed"
          modalLabel=""
          onKeyDown={[Function]}
          onRequestClose={[Function]}
          onRequestSubmit={[Function]}
          open={false}
          passiveModal={true}
          primaryButtonDisabled={false}
        >
          <p>
            Failed to Login to OpenTRV.
          </p>
          <p>
            Please check your email and password was entered correctly.
          </p>
          <p>
            If problem persists please contact support
          </p>
        </Modal>
      </div>,
      "className": "Login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Login__title"
          >
            <h1>
              Login to OpenTRV
            </h1>
          </div>,
          <div
            className="Login__input"
          >
            <div
              className="Login__email-input"
            >
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="login-email"
                invalid={false}
                invalidText=""
                labelText="Email"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter email address"
                required={true}
                type="email"
              />
            </div>
            <div
              className="Login__password-input"
            >
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="login-password"
                invalid={false}
                invalidText=""
                labelText="Password"
                onChange={[Function]}
                onClick={[Function]}
                onKeyPress={[Function]}
                placeholder="Enter password"
                required={true}
                type="password"
              />
            </div>
          </div>,
          <div
            className="Login__buttons"
          >
            <Button
              className="Login__back-button"
              disabled={false}
              href="/"
              iconDescription="Provide icon description if icon is used"
              kind="secondary"
              small={false}
              style={
                Object {
                  "width": "120px",
                }
              }
              tabIndex={0}
              type="button"
            >
              Back
            </Button>
            <Button
              className="Login__login-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="primary"
              onClick={[Function]}
              small={false}
              style={
                Object {
                  "width": "120px",
                }
              }
              tabIndex={0}
              type="button"
            >
              Login
            </Button>
          </div>,
          <Modal
            className="Login__failed-modal"
            iconDescription="close the modal"
            modalHeading="Login Failed"
            modalLabel=""
            onKeyDown={[Function]}
            onRequestClose={[Function]}
            onRequestSubmit={[Function]}
            open={false}
            passiveModal={true}
            primaryButtonDisabled={false}
          >
            <p>
              Failed to Login to OpenTRV.
            </p>
            <p>
              Please check your email and password was entered correctly.
            </p>
            <p>
              If problem persists please contact support
            </p>
          </Modal>,
        ],
        "className": "Login__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Login to OpenTRV
            </h1>,
            "className": "Login__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login to OpenTRV",
            },
            "ref": null,
            "rendered": "Login to OpenTRV",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Login__email-input"
              >
                <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-email"
                  invalid={false}
                  invalidText=""
                  labelText="Email"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Enter email address"
                  required={true}
                  type="email"
                />
              </div>,
              <div
                className="Login__password-input"
              >
                <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-password"
                  invalid={false}
                  invalidText=""
                  labelText="Password"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Enter password"
                  required={true}
                  type="password"
                />
              </div>,
            ],
            "className": "Login__input",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-email"
                  invalid={false}
                  invalidText=""
                  labelText="Email"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Enter email address"
                  required={true}
                  type="email"
                />,
                "className": "Login__email-input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "bx--text__input",
                  "disabled": false,
                  "id": "login-email",
                  "invalid": false,
                  "invalidText": "",
                  "labelText": "Email",
                  "onChange": [Function],
                  "onClick": [Function],
                  "placeholder": "Enter email address",
                  "required": true,
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-password"
                  invalid={false}
                  invalidText=""
                  labelText="Password"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Enter password"
                  required={true}
                  type="password"
                />,
                "className": "Login__password-input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "bx--text__input",
                  "disabled": false,
                  "id": "login-password",
                  "invalid": false,
                  "invalidText": "",
                  "labelText": "Password",
                  "onChange": [Function],
                  "onClick": [Function],
                  "onKeyPress": [Function],
                  "placeholder": "Enter password",
                  "required": true,
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="Login__back-button"
                disabled={false}
                href="/"
                iconDescription="Provide icon description if icon is used"
                kind="secondary"
                small={false}
                style={
                  Object {
                    "width": "120px",
                  }
                }
                tabIndex={0}
                type="button"
              >
                Back
              </Button>,
              <Button
                className="Login__login-button"
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="primary"
                onClick={[Function]}
                small={false}
                style={
                  Object {
                    "width": "120px",
                  }
                }
                tabIndex={0}
                type="button"
              >
                Login
              </Button>,
            ],
            "className": "Login__buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Back",
                "className": "Login__back-button",
                "disabled": false,
                "href": "/",
                "iconDescription": "Provide icon description if icon is used",
                "kind": "secondary",
                "small": false,
                "style": Object {
                  "width": "120px",
                },
                "tabIndex": 0,
                "type": "button",
              },
              "ref": null,
              "rendered": "Back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Login",
                "className": "Login__login-button",
                "disabled": false,
                "iconDescription": "Provide icon description if icon is used",
                "kind": "primary",
                "onClick": [Function],
                "small": false,
                "style": Object {
                  "width": "120px",
                },
                "tabIndex": 0,
                "type": "button",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p>
                Failed to Login to OpenTRV.
              </p>,
              <p>
                Please check your email and password was entered correctly.
              </p>,
              <p>
                If problem persists please contact support
              </p>,
            ],
            "className": "Login__failed-modal",
            "iconDescription": "close the modal",
            "modalHeading": "Login Failed",
            "modalLabel": "",
            "onKeyDown": [Function],
            "onRequestClose": [Function],
            "onRequestSubmit": [Function],
            "open": false,
            "passiveModal": true,
            "primaryButtonDisabled": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Failed to Login to OpenTRV.",
              },
              "ref": null,
              "rendered": "Failed to Login to OpenTRV.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please check your email and password was entered correctly.",
              },
              "ref": null,
              "rendered": "Please check your email and password was entered correctly.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "If problem persists please contact support",
              },
              "ref": null,
              "rendered": "If problem persists please contact support",
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Login__content"
        >
          <div
            className="Login__title"
          >
            <h1>
              Login to OpenTRV
            </h1>
          </div>
          <div
            className="Login__input"
          >
            <div
              className="Login__email-input"
            >
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="login-email"
                invalid={false}
                invalidText=""
                labelText="Email"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter email address"
                required={true}
                type="email"
              />
            </div>
            <div
              className="Login__password-input"
            >
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="login-password"
                invalid={false}
                invalidText=""
                labelText="Password"
                onChange={[Function]}
                onClick={[Function]}
                onKeyPress={[Function]}
                placeholder="Enter password"
                required={true}
                type="password"
              />
            </div>
          </div>
          <div
            className="Login__buttons"
          >
            <Button
              className="Login__back-button"
              disabled={false}
              href="/"
              iconDescription="Provide icon description if icon is used"
              kind="secondary"
              small={false}
              style={
                Object {
                  "width": "120px",
                }
              }
              tabIndex={0}
              type="button"
            >
              Back
            </Button>
            <Button
              className="Login__login-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="primary"
              onClick={[Function]}
              small={false}
              style={
                Object {
                  "width": "120px",
                }
              }
              tabIndex={0}
              type="button"
            >
              Login
            </Button>
          </div>
          <Modal
            className="Login__failed-modal"
            iconDescription="close the modal"
            modalHeading="Login Failed"
            modalLabel=""
            onKeyDown={[Function]}
            onRequestClose={[Function]}
            onRequestSubmit={[Function]}
            open={false}
            passiveModal={true}
            primaryButtonDisabled={false}
          >
            <p>
              Failed to Login to OpenTRV.
            </p>
            <p>
              Please check your email and password was entered correctly.
            </p>
            <p>
              If problem persists please contact support
            </p>
          </Modal>
        </div>,
        "className": "Login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Login__title"
            >
              <h1>
                Login to OpenTRV
              </h1>
            </div>,
            <div
              className="Login__input"
            >
              <div
                className="Login__email-input"
              >
                <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-email"
                  invalid={false}
                  invalidText=""
                  labelText="Email"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Enter email address"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="Login__password-input"
              >
                <TextInput
                  className="bx--text__input"
                  disabled={false}
                  id="login-password"
                  invalid={false}
                  invalidText=""
                  labelText="Password"
                  onChange={[Function]}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Enter password"
                  required={true}
                  type="password"
                />
              </div>
            </div>,
            <div
              className="Login__buttons"
            >
              <Button
                className="Login__back-button"
                disabled={false}
                href="/"
                iconDescription="Provide icon description if icon is used"
                kind="secondary"
                small={false}
                style={
                  Object {
                    "width": "120px",
                  }
                }
                tabIndex={0}
                type="button"
              >
                Back
              </Button>
              <Button
                className="Login__login-button"
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="primary"
                onClick={[Function]}
                small={false}
                style={
                  Object {
                    "width": "120px",
                  }
                }
                tabIndex={0}
                type="button"
              >
                Login
              </Button>
            </div>,
            <Modal
              className="Login__failed-modal"
              iconDescription="close the modal"
              modalHeading="Login Failed"
              modalLabel=""
              onKeyDown={[Function]}
              onRequestClose={[Function]}
              onRequestSubmit={[Function]}
              open={false}
              passiveModal={true}
              primaryButtonDisabled={false}
            >
              <p>
                Failed to Login to OpenTRV.
              </p>
              <p>
                Please check your email and password was entered correctly.
              </p>
              <p>
                If problem persists please contact support
              </p>
            </Modal>,
          ],
          "className": "Login__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Login to OpenTRV
              </h1>,
              "className": "Login__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login to OpenTRV",
              },
              "ref": null,
              "rendered": "Login to OpenTRV",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Login__email-input"
                >
                  <TextInput
                    className="bx--text__input"
                    disabled={false}
                    id="login-email"
                    invalid={false}
                    invalidText=""
                    labelText="Email"
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder="Enter email address"
                    required={true}
                    type="email"
                  />
                </div>,
                <div
                  className="Login__password-input"
                >
                  <TextInput
                    className="bx--text__input"
                    disabled={false}
                    id="login-password"
                    invalid={false}
                    invalidText=""
                    labelText="Password"
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Enter password"
                    required={true}
                    type="password"
                  />
                </div>,
              ],
              "className": "Login__input",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <TextInput
                    className="bx--text__input"
                    disabled={false}
                    id="login-email"
                    invalid={false}
                    invalidText=""
                    labelText="Email"
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder="Enter email address"
                    required={true}
                    type="email"
                  />,
                  "className": "Login__email-input",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "bx--text__input",
                    "disabled": false,
                    "id": "login-email",
                    "invalid": false,
                    "invalidText": "",
                    "labelText": "Email",
                    "onChange": [Function],
                    "onClick": [Function],
                    "placeholder": "Enter email address",
                    "required": true,
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <TextInput
                    className="bx--text__input"
                    disabled={false}
                    id="login-password"
                    invalid={false}
                    invalidText=""
                    labelText="Password"
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Enter password"
                    required={true}
                    type="password"
                  />,
                  "className": "Login__password-input",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "bx--text__input",
                    "disabled": false,
                    "id": "login-password",
                    "invalid": false,
                    "invalidText": "",
                    "labelText": "Password",
                    "onChange": [Function],
                    "onClick": [Function],
                    "onKeyPress": [Function],
                    "placeholder": "Enter password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  className="Login__back-button"
                  disabled={false}
                  href="/"
                  iconDescription="Provide icon description if icon is used"
                  kind="secondary"
                  small={false}
                  style={
                    Object {
                      "width": "120px",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  Back
                </Button>,
                <Button
                  className="Login__login-button"
                  disabled={false}
                  iconDescription="Provide icon description if icon is used"
                  kind="primary"
                  onClick={[Function]}
                  small={false}
                  style={
                    Object {
                      "width": "120px",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  Login
                </Button>,
              ],
              "className": "Login__buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Back",
                  "className": "Login__back-button",
                  "disabled": false,
                  "href": "/",
                  "iconDescription": "Provide icon description if icon is used",
                  "kind": "secondary",
                  "small": false,
                  "style": Object {
                    "width": "120px",
                  },
                  "tabIndex": 0,
                  "type": "button",
                },
                "ref": null,
                "rendered": "Back",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Login",
                  "className": "Login__login-button",
                  "disabled": false,
                  "iconDescription": "Provide icon description if icon is used",
                  "kind": "primary",
                  "onClick": [Function],
                  "small": false,
                  "style": Object {
                    "width": "120px",
                  },
                  "tabIndex": 0,
                  "type": "button",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <p>
                  Failed to Login to OpenTRV.
                </p>,
                <p>
                  Please check your email and password was entered correctly.
                </p>,
                <p>
                  If problem persists please contact support
                </p>,
              ],
              "className": "Login__failed-modal",
              "iconDescription": "close the modal",
              "modalHeading": "Login Failed",
              "modalLabel": "",
              "onKeyDown": [Function],
              "onRequestClose": [Function],
              "onRequestSubmit": [Function],
              "open": false,
              "passiveModal": true,
              "primaryButtonDisabled": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Failed to Login to OpenTRV.",
                },
                "ref": null,
                "rendered": "Failed to Login to OpenTRV.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please check your email and password was entered correctly.",
                },
                "ref": null,
                "rendered": "Please check your email and password was entered correctly.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "If problem persists please contact support",
                },
                "ref": null,
                "rendered": "If problem persists please contact support",
                "type": "p",
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
